src/main/groovy/org/jenkinsci/plugins/pipeline/config/parser/JSONParser.groovy

```diff
@@ -223,11 +223,15 @@ class JSONParser {
     }
 
     public @CheckForNull ConfigASTValue parseValue(JSONObject o) {
-        if (o.getBoolean("isConstant")) {
-            return ConfigASTValue.fromConstant(o.get("value"), o)
-        } else {
-            return ConfigASTValue.fromGString(o.getString("value"), o)
+        // TODO: Probably need better type-checking? Not sure yet.
+        if (o.getString("value") != null) {
+            String valString = o.getString("value")
+            if (valString.contains('$class')) {
```

Comment:
One of the places I'm concerned with special casing - I need some way to tell when a value is a string representation of an expression, and so far, that's the best I've got...but it won't work with some other things (like passing a list or map to a step on purpose) so...hrm.